dashboard/app/models/user_level.rb

```diff
@@ -109,12 +109,14 @@ def locked?(stage)
   end
 
   def self.update_lockable_state(user_id, level_id, script_id, locked, readonly_answers)
-    user_level = UserLevel.find_by(user_id: user_id, level_id: level_id, script_id: script_id)
+    user_level = UserLevel.find_or_initialize_by(
+      user_id: user_id,
+      level_id: level_id,
+      script_id: script_id
+    )
 
     # no need to create a level if it's just going to be locked
-    return if !user_level && locked
-
-    user_level ||= UserLevel.create(user_id: user_id, level_id: level_id, script_id: script_id)
```

Comment:
Note that, rather than combining the `create` with the above `find_or_initialize_by`, this could have alternatively been `UserLevel.new` to remove the DB write.